src/components/network/network-explorer.js

```diff
@@ -59,22 +71,61 @@ const NetworkExplorer = ({network, onVoltageLevelClick}) => {
 
     useEffect(() => {
         if (currentVoltageLevel !== null) {
-            onVoltageLevelClick(currentVoltageLevel.id, currentVoltageLevel.name);
+            onVoltageLevelDisplayClick(currentVoltageLevel.id, currentVoltageLevel.name);
         }
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [useName]);
 
-    function onClickHandler(index) {
-        if (onVoltageLevelClick !== null) {
+    function handleClick(index) {
+        setVoltageLevelMenuIndex(index);
+    }
+
+    function handleClose() {
+        setVoltageLevelMenuIndex(-1);
+    }
+
+    function onDisplayClickHandler(index = voltageLevelMenuIndex) {
+        if (onVoltageLevelDisplayClick !== null) {
             const vl = filteredVoltageLevels[index];
-            onVoltageLevelClick(vl.id);
+            onVoltageLevelDisplayClick(vl.id);
             setCurrentVoltageLevel(vl);
         }
+        handleClose();
     }
 
+    function onFocusClickHandler() {
+        if (onVoltageLevelFocusClick !== null) {
+            const vl = filteredVoltageLevels[voltageLevelMenuIndex];
+            onVoltageLevelFocusClick(vl.id);
+        }
+        handleClose();
+    }
+
+    const voltagelevelInfo = index => {
+        if (network.getSubstation(filteredVoltageLevels[index].substationId)) {
+            return network.getSubstation(filteredVoltageLevels[index].substationId).countryName + " - " + filteredVoltageLevels[index].nominalVoltage + " kV";
```

Comment:
maybe put the voltage first as it is always short. If we ellipsize it will still be visible